WooCommerce/WooCommerceUITests/Screens/Login/PrologueScreen.swift

```diff
@@ -1,20 +1,27 @@
-import Foundation
+import ScreenObject
 import XCTest
 
 private struct ElementStringIDs {
     static let continueButton = "Prologue Continue Button"
     static let siteAddressButton = "Prologue Self Hosted Button"
 }
 
-final class PrologueScreen: BaseScreen {
-    private let continueButton: XCUIElement
-    private let siteAddressButton: XCUIElement
+final class PrologueScreen: ScreenObject {
+    private let continueButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons[ElementStringIDs.continueButton]
+    }
+    private let siteAddressButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons[ElementStringIDs.siteAddressButton]
+    }
 
-    init() {
-        continueButton = XCUIApplication().buttons[ElementStringIDs.continueButton]
-        siteAddressButton = XCUIApplication().buttons[ElementStringIDs.siteAddressButton]
+    private var continueButton: XCUIElement { expectedElement }
+    private var siteAddressButton: XCUIElement { siteAddressButtonGetter(app) }
 
-        super.init(element: continueButton)
+    init(app: XCUIApplication = XCUIApplication()) throws {
+        try super.init(
+            expectedElementGetter: { $0.buttons[ElementStringIDs.continueButton] },
```

Comment:
Presumably this replaces `isLoaded`? I like that a lot

Reply:
Thanks 😄 

This is the code that `isLoaded` uses to fetch the element to use for its check under the hood. See step one [here](https://github.com/Automattic/ScreenObject/blob/df8ee4983e674b583b546bb0640ec83428be2465/Sources/ScreenObject/ScreenObject.swift#L28) and step two [here](https://github.com/Automattic/ScreenObject/blob/df8ee4983e674b583b546bb0640ec83428be2465/Sources/ScreenObject/ScreenObject.swift#L18).